tests/vmi_multus_test.go

```diff
@@ -846,6 +848,45 @@ var _ = Describe("SRIOV", func() {
 			// it's hard to match them.
 		})
 
+		Context("VLANed SR-IOV network", func() {
+			var vlanedVMI1 *v1.VirtualMachineInstance
+			cidrVlaned1 := "192.168.0.1/24"
+			cidrVlaned2 := "192.168.0.2/24"
+			cidrFlat := "192.168.1.1/24"
+			BeforeEach(func() {
+				createNetwork("sriov-vlan", tests.NamespaceTestDefault, sriovConfVlanCRD)
+				createNetwork("sriov-no-vlan", tests.NamespaceTestDefault, sriovConfNoVlanCrd)
+				vlanedVMI1 = createSriovVMI([]string{"sriov-vlan"})
+
+				// manually configure IP/link on sriov interfaces because there is
+				// no DHCP server to serve the address to the guest
+				configInterface(vlanedVMI1, "eth1", cidrVlaned1, "#")
+			})
+
+			AfterEach(func() {
+				deleteNetwork("sriov-vlan", tests.NamespaceTestDefault)
+				deleteNetwork("sriov-no-vlan", tests.NamespaceTestDefault)
+			})
+
+			It("[test_id:3958]should be able to ping between to VMs with the same VLAN over SRIOV network", func() {
+				vlanedVMI2 := createSriovVMI([]string{"sriov-vlan"})
+				configInterface(vlanedVMI2, "eth1", cidrVlaned2, "#")
+
+				By("pinging between vlanedVMI1 and vlanedVMI2 over vlan")
+				err = pingVirtualMachine(vlanedVMI1, cidrToIP(cidrVlaned2), "#")
+				Expect(err).ToNot(HaveOccurred(), "VMI %s should be reachable", vlanedVMI2.Name, err)
+			})
+
+			It("[test_id:3958]should NOT be able to ping between Vlaned VM and a flat VM", func() {
```

Comment:
I think you should create all 3 VMI/s and then perform multiple asserts. The cost of repeating the same action is not worth the test isolation.

BTW, where this test ID came from?